wordpress-mobile/WordPress-Android
# Add Jetpack badge to My Site

This PR adds the "Jetpack powered" badge to My Site (~menu tab~, home tab and the version without tabs)
It also introduces the Jetpack badge component itself, which is implemented as a `MaterialButton`.

| Home Tab | Dashboard without tabs |
| --- | --- |
| ![home_badge_light](https://user-images.githubusercontent.com/4588074/178552554-e5b70029-b86f-4e47-8ffd-74cccb284afc.png) | ![dashboard_badge_night](https://user-images.githubusercontent.com/4588074/178690851-9af5c466-eab3-4c61-acea-7738f1fa3f33.png) |

⚠️ **note** There's a [64dp bottom padding](https://github.com/wordpress-mobile/WordPress-Android/blob/198081447966a7acc182b8b3c3b86cc60797c413/WordPress/src/main/res/layout/my_site_tab_fragment.xml#L29) set on the recycler view which makes the badge appear upper compared to the original designs.

To test:
1. **WordPress app**
   - Build/run this branch
   - Go to My Site (by logging in) and select the **Home** tab
   - **Expect** the "Jetpack powered" badge to **appear** at the bottom of the home tab (see screenshot above)
   - Select the **Menu** tab
   - **Expect** the "Jetpack powered" badge to **not appear** at the bottom of the menu
   - Go to Me → App Settings → Debug Settings, disable `my_site_dashboard_tabs` & tap `RESTART THE APP`
   - **Expect** the "Jetpack powered" badge to **appear** at the bottom of the My Site Dashboard
2. **Jetpack app**
   - Repeat the above steps and **expect** the Jetpack powered badge to **never appear**

## Regression Notes
1. Potential unintended areas of impact
   Low impact on My site.

4. What I did to test those areas of impact (or what existing automated tests I relied on)
   Manual Testing on different screen sizes.

6. What automated tests I added (or what prevented me from doing so)
   Updated 1 existing unit test and added 2 new ones in `MySiteViewModelTest`.

PR submission checklist:

      Not released.
